schema.h

```diff
@@ -47,53 +62,77 @@ struct User : yoda::Padawan {
   template <typename A>
   void serialize(A& ar) {
     Padawan::serialize(ar);
-    ar(CEREAL_NVP(score));
+    ar(CEREAL_NVP(level), CEREAL_NVP(score));
   }
 };
 
-struct UIDTokenPair : yoda::Padawan {
-  UID uid = UID::INVALID;
+// AuthKey structure defines generic authentication key, which consists of a pair of `std::string`s.
+// `key1` and `key2` may be used in different ways, depending on the authentication type.
+// For `type = AUTH_TYPE::IOS`:
+//  - key1 = Device ID
+//  - key2 = Application key, a random number, generated once in a lifetime in iOS application on its first
+//  launch.
+struct AuthKey {
+  std::string key1 = "";
+  std::string key2 = "";
+  AUTH_TYPE type = AUTH_TYPE::UNDEFINED;
+
+  AuthKey() = default;
+  AuthKey(const std::string& key1, const std::string& key2, AUTH_TYPE type)
+      : key1(key1), key2(key2), type(type) {}
+  size_t Hash() const { return std::hash<std::string>()(key1) ^ std::hash<std::string>()(key2); }
+  bool operator==(const AuthKey& rhs) const { return key1 == rhs.key1 && key2 == rhs.key2; }
```

Comment:
While I see the reason to not include `type` in `Hash()`, it should certainly be here in `operator==`.

(In `Hash()`, it's OK to only keep the random ID, if you ask me.)